src/input-nested-multi-select/input-nested-multi-select.tsx

```diff
@@ -106,11 +106,17 @@ export const InputNestedMultiSelect = <V1, V2, V3>({
             );
 
             if (selectedCount < selectableOptionKeyPaths.length) {
-                newKeyPaths = selectableOptionKeyPaths;
+                newKeyPaths = [
+                    ...new Map(
+                        [...selectedKeyPaths, ...selectableOptionKeyPaths].map(
+                            (k) => [k.join("-"), k]
+                        )
+                    ).values(),
+                ];
```

Comment:
can explain this part?

was expecting `selectableOptionKeyPaths` to be directly concatenated to the `newKeyPaths` array

Reply:
`...new Map().values()` try to get the value unique.

Scenario:
1. Select first one category L2 item which key path is `["100", "200"]`;
2. User select the checkbox on L1 category, `selectableOptionKeyPaths` included the `["100", "200"]`.
3. `selectedKeyPaths` also included `["100", "200"]` which step 1 user already selected.
4. `new Map()` try to remove the duplication. similar the `new Set` method.

Reply:
ok. this should be fine for most cases. will take this implementation.

in the previous line `newKeyPaths` already removed the subitems, so the duplication shouldn't happen